File: hometask_06/src/main/java/com/example/hometask_06/ContactClass.java
package com.example.hometask_06;

import com.example.hometask_06.database.ContactEntity;

public class ContactClass {
    private String name;
    private String numberOrEmail;
    boolean isEmail ;

    public ContactClass(String name, String numberOrEmail, boolean isEmail) {
        this.name = name;
        this.numberOrEmail = numberOrEmail;
        this.isEmail = isEmail;
    }

    public String getName() {
        return name;
    }

    public String getNumberOrEmail() {
        return numberOrEmail;
    }

    public boolean isEmail() {
        return isEmail;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setNumberOrEmail(String numberOrEmail) {
        this.numberOrEmail = numberOrEmail;
    }

    public void setEmail(boolean email) {
        isEmail = email;
    }

    static ContactEntity createEntity(ContactClass contact) {
        ContactEntity entity = new ContactEntity(contact.getName(), contact.getNumberOrEmail() , contact.isEmail()) ;
        return entity ;
    }
}

File: hometask_04_kotlincontacts/src/main/java/com/example/hometask_04_kotlincontacts/ContactClass.kt
package com.example.hometask_04_kotlincontacts

class ContactClass (private val config: Builder) {

    lateinit var name : String
    lateinit var numberOrEmail : String
    var isEmail : Boolean = false

    init {
        this.name = Builder.name
        this.numberOrEmail = Builder.numberOrEmail
        this.isEmail = Builder.isEmail
    }

    companion object Builder {
        lateinit var name : String
        lateinit var numberOrEmail : String
        var isEmail : Boolean = false

        fun setName(name : String) : Builder {
            this.name = name
            return this
        }

        fun setNumberOrEmail (info : String) : Builder {
            this.numberOrEmail = info
            return this
        }

        fun itIsEmail() : Builder {
            this.isEmail = true
            return this
        }

        fun build() : ContactClass {
            val newContact = ContactClass(this)
            return newContact
        }
    }
}